### PR TITLE
Add withEndpointConfiguration() option for S3Client

### PR DESCRIPTION
**Pull request by helenhua**
Goal: Allow user to add an optional <endpoint> field in their AWS S3 profile configuration so that one can upload files to a S3 compatible cloud storage service

AwsS3ServiceImpl.java: Check for endpoint and region, and if both are given, then we can use withEndpointConfiguration() to connect to a S3 compatible storage

### CODE REVIEW

File: src/main/java/org/craftercms/studio/impl/v2/service/aws/s3/AwsS3ServiceImpl.java [hunk 1]
```diff
@@ -61,12 +61,19 @@ public void setPartSize(final int partSize) {
...
+    /** 
+    * Add withEndpointConfiguration() to direct requests to a S3 compatible storage service
+    */
     protected AmazonS3 getS3Client(S3Profile profile) {
         AmazonS3ClientBuilder builder = AmazonS3ClientBuilder.standard()
             .withCredentials(profile.getCredentialsProvider());
-        if(StringUtils.isNotEmpty(profile.getRegion())) {
+            
+        if(StringUtils.isNotEmpty(profile.getEndpoint()) && StringUtils.isNotEmpty(profile.getRegion())){
+            builder.withEndpointConfiguration( new AmazonS3ClientBuilder.EndpointConfiguration(profile.getEndpoint(), profile.getRegion()));
```
Comment (avasquez614) [inline src/main/java/org/craftercms/studio/impl/v2/service/aws/s3/AwsS3ServiceImpl.java]: there's an unnecessary space between the `(` and the `new` keyword.

File: src/main/java/org/craftercms/studio/impl/v2/service/aws/s3/AwsS3ServiceImpl.java [hunk 2]
```diff
@@ -61,12 +61,19 @@ public void setPartSize(final int partSize) {
...
 
+    /** 
+    * Add withEndpointConfiguration() to direct requests to a S3 compatible storage service
+    */
     protected AmazonS3 getS3Client(S3Profile profile) {
         AmazonS3ClientBuilder builder = AmazonS3ClientBuilder.standard()
             .withCredentials(profile.getCredentialsProvider());
-        if(StringUtils.isNotEmpty(profile.getRegion())) {
+            
+        if(StringUtils.isNotEmpty(profile.getEndpoint()) && StringUtils.isNotEmpty(profile.getRegion())){
```
Comment (avasquez614) [inline src/main/java/org/craftercms/studio/impl/v2/service/aws/s3/AwsS3ServiceImpl.java]: please add a space after the `if`, like `if (`.

File: src/main/java/org/craftercms/studio/impl/v2/service/aws/s3/AwsS3ServiceImpl.java [hunk 3]
```diff
@@ -61,12 +61,19 @@ public void setPartSize(final int partSize) {
...
+            builder.withEndpointConfiguration( new AmazonS3ClientBuilder.EndpointConfiguration(profile.getEndpoint(), profile.getRegion()));
+        } else if(StringUtils.isNotEmpty(profile.getRegion())) {
```
Comment (avasquez614) [inline src/main/java/org/craftercms/studio/impl/v2/service/aws/s3/AwsS3ServiceImpl.java]: please add a space after the `if`, like `if (`.

**Review by avasquez614 — CHANGES_REQUESTED** [review]

**Review by avasquez614 — APPROVED** [review]

**Review by jdrossl — APPROVED** [review]

**Comment by sumerjabri** [comment]
Waiting on https://github.com/craftercms/commons/pull/159

**Pull request merged**